numpy/ma/core.py

```diff
@@ -3316,13 +3316,10 @@ def _scalar_heuristic(arr, elem):
                                 "heterogeneous fill_value and setting "
                                 f"all to {dout._fill_value[0]!s}.",
                                 stacklevel=2)
-                        dout._fill_value = dout._fill_value.flat[0]
-                    elif not hasattr(dout._fill_value, 'ndim'):
-                        # only initialize to empty array if it dout does not 
-                        # have ._fill_value before. If it has, do not change 
-                        # the original behavior
-                        dout._fill_value = np.empty((), dtype)
-                        dout._fill_value[()] = self._fill_value[indx]
+                    # need to use squeeze to ensure the result is an array 
+                    # not a scalar. If the _fill_value is like a str type,
+                    # then just make sure we turn the _fill_value to a 0d array
+                    dout._fill_value = dout._fill_value.flat[0:1].squeeze(axis=0)
```

Comment:
Needs indenting by one

Reply:
But if we indent by one, then it does not solve for the case when dtype.ndim is 0 though

Reply:
I don't believe the problem exists when it's 0

Reply:
The problem was caused by the use of `.flat`

Reply:
Ah I see! Thank you!